lib/Github/Api/Repository/Checks.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Github\Api\Repository;
+
+use Github\Api\AbstractApi;
+use Github\Api\AcceptHeaderTrait;
+use Github\Exception\MissingArgumentException;
+
+/**
+ * @link   https://developer.github.com/v3/checks/
+ *
+ * @author Zack Galbreath <zack.galbreath@kitware.com>
+ */
+class Checks extends AbstractApi
+{
+    use AcceptHeaderTrait;
+
+    public function configure()
+    {
+        $this->acceptHeaderValue = 'application/vnd.github.antiope-preview+json';
+
+        return $this;
+    }
+
+    /**
+     * @link https://developer.github.com/v3/checks/runs/#create-a-check-run
+     *
+     * @param string $username
+     * @param string $repository
+     * @param array  $params
+     *
+     * @throws MissingArgumentException
+     *
+     * @return array
+     */
+    public function create($username, $repository, array $params = [])
+    {
+        if (!isset($params['name'], $params['head_sha'])) {
```

Comment:
Maybe we can just set the correct acceptHeader in the create and update method. This way the user is not required to call the configure method and we have 1 method less to worry about to keep BC, we can just remove the acceptHeader set in these methods when the api is out of preview mode. See example: https://github.com/KnpLabs/php-github-api/blob/8363f9567e28ce3e4a22d2c2cc05ac7c8497e008/lib/Github/Api/Repo.php#L624

What do you think?

Reply:
👍